activemq-broker/src/main/java/org/apache/activemq/broker/region/RegionBroker.java

```diff
@@ -781,23 +781,8 @@ public boolean sendToDeadLetterQueue(ConnectionContext context, MessageReference
                                 return false;
                             }
 
-                            // message may be inflight to other subscriptions so do not modify
-                            message = message.copy();
-                            long dlqExpiration = deadLetterStrategy.getExpiration();
-                            if (dlqExpiration > 0) {
-                                dlqExpiration += System.currentTimeMillis();
-                            } else {
-                                stampAsExpired(message);
-                            }
-                            message.setExpiration(dlqExpiration);
-                            if (!message.isPersistent()) {
-                                message.setPersistent(true);
-                                message.setProperty("originalDeliveryMode", "NON_PERSISTENT");
-                            }
-                            if (poisonCause != null) {
-                                message.setProperty(ActiveMQMessage.DLQ_DELIVERY_FAILURE_CAUSE_PROPERTY,
-                                        poisonCause.toString());
-                            }
+                            message = prepareMessageForDeadLetterQueue(message, deadLetterStrategy, poisonCause);
```

Comment:
Well, while the method is in fact descriptive about what it is doing, it actually makes it harder to understand what you changed because by copying it all to a new method it's not clear what exactly the difference is from the old code without looking at it carefully as the entire method shows up as a new change which is the biggest reason why I made the comment because i as staring at it for a while to even figure out what was different.

Reply:
Makes sense, I've removed the `prepareMessageForDeadLetterQueue` method.